optimum/bettertransformer/models/encoder_models.py

```diff
@@ -17,6 +17,114 @@
 from .base import BetterTransformerBaseLayer
 
 
+class TapasLayerBetterTransformer(BetterTransformerBaseLayer):
+    def __init__(self, tapas_layer, config):
+        r"""
+        A simple conversion of the TAPAS layer to its `BetterTransformer` implementation.
+
+        Args:
+            tapas_layer (`torch.nn.Module`):
+                The original TAPAS Layer where the weights needs to be retrieved.
+        """
+        super().__init__(config)
+        # In_proj layer
+        self.in_proj_weight = nn.Parameter(
+            torch.cat(
+                [
+                    tapas_layer.attention.query.weight,
```

Comment:
looking at the Tapas implementation, https://github.com/huggingface/transformers/blob/28247e78819ab9756b81f8df39611c333d099400/src/transformers/models/tapas/modeling_tapas.py#L442 , I think we need here and in the rest `tapas_layer.attention.self.query.weight`, that is why the test is failing!

Reply:
Oh........ thanks you for providing me with your insights!